### PR TITLE
feat(pipeline): set up ci/cd pipeline

### PR DESCRIPTION
**Pull request by simonebeets**
* Ran eslint check
* All tests passed locally

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: 2
+updates:
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    target-branch: develop
+    schedule:
+      interval: weekly
```
Comment (thiren) [inline .github/dependabot.yml:8]: lets change the interval to monthly

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    open-pull-requests-limit: 10
+  # Maintain dependencies for npm
+  - package-ecosystem: "npm"
+    directory: "/"
+    target-branch: develop
+    schedule:
+      interval: weekly
```
Comment (thiren) [inline .github/dependabot.yml:15]: lets change the interval to monthly

File: .github/workflows/ci-codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: "CodeQL Analysis"
+
+on:
+  push:
+    branches: [develop]
```
Comment (thiren) [inline .github/workflows/ci-codeql-analysis.yml:5]: should be: `branches: [master, develop]`

File: .github/workflows/ci-npm-publish.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      # Skip post-install scripts here, as a malicious
+      # script could steal NODE_AUTH_TOKEN.
+      - name: Install NPM dependencies
+        run: npm install --ignore-scripts --production
+
+      - name: Run NPM publish
+        run: npm publish
```
Comment (thiren) [inline .github/workflows/ci-npm-publish.yml:51]: should be: `run: npm publish --access public`

File: .github/workflows/ci-npm-publish.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  publish-npm:
+    name: NPM Publish
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (thiren) [inline .github/workflows/ci-npm-publish.yml:43]: set the node version to 14

File: .github/workflows/ci-npm-publish.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      mongo:
+        image: mongo:4.4
+        ports:
+          - 27017:27017
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (thiren) [inline .github/workflows/ci-npm-publish.yml:21]: set the node version to 14

File: .github/workflows/ci-build.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  pack:
+    name: NPM Pack
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (thiren) [inline .github/workflows/ci-build.yml:51]: set the node version to 14

File: .github/workflows/ci-build.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      mongo:
+        image: mongo:4.4
+        ports:
+          - 27017:27017
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (thiren) [inline .github/workflows/ci-build.yml:29]: set the node version to 14

**Review by thiren — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (simonebeets) [reply .github/dependabot.yml]: done

File: .github/dependabot.yml (hunk 2 above)
Comment (simonebeets) [reply .github/dependabot.yml]: done

File: .github/workflows/ci-codeql-analysis.yml (hunk 3 above)
Comment (simonebeets) [reply .github/workflows/ci-codeql-analysis.yml]: done, we had an issue on steps-mongo when master was included so I had to remove it and that is why I left it out here, but added it back.

File: .github/workflows/ci-npm-publish.yml (hunk 4 above)
Comment (simonebeets) [reply .github/workflows/ci-npm-publish.yml]: done

File: .github/workflows/ci-npm-publish.yml (hunk 5 above)
Comment (simonebeets) [reply .github/workflows/ci-npm-publish.yml]: done

File: .github/workflows/ci-npm-publish.yml (hunk 6 above)
Comment (simonebeets) [reply .github/workflows/ci-npm-publish.yml]: done

File: .github/workflows/ci-build.yml (hunk 7 above)
Comment (simonebeets) [reply .github/workflows/ci-build.yml]: done

File: .github/workflows/ci-build.yml (hunk 8 above)
Comment (simonebeets) [reply .github/workflows/ci-build.yml]: done

**Review by thiren — APPROVED** [review]

**Pull request merged**